Use native ESPHome components, instead of the adf_pipeline/esphome_audio

migo007 opened the pull request:
As all used components are now also available upstream in ESPHome, this pull request tries to remove the adf_pipeline and esphome_audio as external dependency in favor of native components. It's compatible with the latest esp-idf (>5). It does not implement version pinning.
The PR also ports some functions from home assistant voice pe configuration:
[https://github.com/esphome/home-assistant-voice-pe/blob/dev/home-assistant-voice.yaml](https://github.com/esphome/home-assistant-voice-pe/blob/dev/home-assistant-voice.yaml)
- Timer support 
- Uninterrupted media playback (ducking on announcements)
- wake sounds

Tested on my Esparagus Echo Duo Rev. B

anabolyc commented:
Thanks @migo007, let me test it

anabolyc commented:
@migo007, it works really well. I think it even wants to work with rev A board, that have both speaker and mic on the same I2S bus (it was failing with adf_pipeline). Can you please update the `echo-solo-b-voice-assist.yaml` file as well, and I'll merge it.

migo007 commented:
@anabolyc Thanks :) ! I updated the solo. For the solo, I changed the speaker mixer to mono mode. I would still recommend testing, if that works as I understood it.

anabolyc commented:
Tested with solo, works really well!
